grakn-graph/src/main/java/ai/grakn/graph/internal/ElementFactory.java

```diff
@@ -60,93 +57,58 @@
     }
 
     // ------------------------------------------- Building Castings  --------------------------------------------------
-    CastingImpl buildCasting(Vertex v, RoleType type){
-        return buildCasting(v, Optional.of(type));
-    }
-    private CastingImpl buildCasting(Vertex v, Optional<RoleType> type){
+    CastingImpl buildCasting(Vertex v, Optional<RoleType> type){
         return trackConcept(new CastingImpl(graknGraph, v, type));
     }
 
-    // -------------------------------------------- Building Types  ----------------------------------------------------
-    private TypeImpl buildType(Vertex v, Optional<Type> type, Optional<Boolean> isImplicit){
-        return trackConcept(new TypeImpl<>(graknGraph, v, type, isImplicit));
-    }
-
     // ---------------------------------------- Building Resource Types  -----------------------------------------------
-    <V> ResourceTypeImpl<V> buildResourceType(Vertex v, ResourceType<V> type, ResourceType.DataType<V> dataType, Boolean isUnique){
-        return buildResourceType(v, Optional.of(type), Optional.of(dataType), Optional.of(isUnique));
-    }
-    private <V> ResourceTypeImpl<V> buildResourceType(Vertex v, Optional<ResourceType<V>> type, Optional<ResourceType.DataType<V>> dataType, Optional<Boolean> isUnique){
+    <V> ResourceTypeImpl<V> buildResourceType(Vertex v, Optional<ResourceType<V>> type, Optional<ResourceType.DataType<V>> dataType, Optional<Boolean> isUnique){
```

Comment:
Can we remove the optionals from these methods now? I quickly pulled down the code and in all usages I saw they are never empty.

It looks like they used to be necessary because `buildConcept` called them.

It might also now be possible to remove them from the constructors by providing two constructors: one with all arguments, one with none (except graph and vertex).